### PR TITLE
New user definition entry

### PR DESCRIPTION
**Pull request by anderson-joyle**
Issue https://github.com/microsoft/Power-Fx/issues/2186.

### CODE REVIEW

**Comment by LucGenetier** [comment]
✅ No public API change.

**Comment by LucGenetier** [comment]
✅ No public API change.

**Comment by LucGenetier** [comment]
✅ No public API change.

**Comment by LucGenetier** [comment]
✅ No public API change.

File: src/libraries/Microsoft.PowerFx.Interpreter/RecalcEngine.cs [hunk 1]
```diff
@@ -381,5 +385,44 @@ public void UpdateSupportedFunctions(SymbolTable s)
         {
             SupportedFunctions = s;
         }
+
+        /// <summary>
+        /// Add a set of user-defined formulas and functions to the engine.
+        /// </summary>
+        /// <param name="script">Script containing user defined functions and/or named formulas.</param>
+        /// <param name="parseCulture">Locale to parse user defined script.</param>
+        /// <param name="symbolTable">Extra SymbolTable if needed.</param>
```
Comment (lesaltzm) [inline src/libraries/Microsoft.PowerFx.Interpreter/RecalcEngine.cs:394]: When would this be useful?

**Comment by LucGenetier** [comment]
✅ No public API change.

File: src/libraries/Microsoft.PowerFx.Interpreter/RecalcEngine.cs (hunk 1 above)
Comment (anderson-joyle) [reply src/libraries/Microsoft.PowerFx.Interpreter/RecalcEngine.cs:394]: This was part of the original implementation.

File: src/libraries/Microsoft.PowerFx.Interpreter/RecalcEngine.cs [hunk 2]
```diff
@@ -381,5 +385,44 @@ public void UpdateSupportedFunctions(SymbolTable s)
...
         }
+
+        /// <summary>
+        /// Add a set of user-defined formulas and functions to the engine.
+        /// </summary>
+        /// <param name="script">Script containing user defined functions and/or named formulas.</param>
+        /// <param name="parseCulture">Locale to parse user defined script.</param>
+        /// <param name="symbolTable">Extra SymbolTable if needed.</param>
+        /// <param name="onUpdate">Function to be called when update is triggered.</param>
+        public void AddUserDefinitions(string script, CultureInfo parseCulture = null, ReadOnlySymbolTable symbolTable = null, Action<string, FormulaValue> onUpdate = null)
```
[the remark this answers is not in the document]
Comment (anderson-joyle) [reply src/libraries/Microsoft.PowerFx.Interpreter/RecalcEngine.cs:396]: Added.

**Comment by LucGenetier** [comment]
✅ No public API change.

**Review by lesaltzm — APPROVED** [review]

**Comment by LucGenetier** [comment]
✅ No public API change.

**Pull request merged**